### PR TITLE
Workaround docker-pull issues with examples/cluster-dns

### PR DESCRIPTION
**Pull request by sttts**
The cluster-dns example is used as conformance test in `tests/e2e/examples.go`. The dns-frontend image often triggers a docker-pull freeze (which is independent of k8s), but block conformance testing.

This PR
- changes the base image to `python:2.7-slim` which is only 30% the original size
- changes the docker registry to gcr.io

**NOTE**: tests will fail until the images are pushed to gcr.io. The Dockerfiles are in `examples/cluster-dns/images`

### CODE REVIEW

File: examples/cluster-dns/dns-backend-rc.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
     spec:
       containers:
         - name: dns-backend
-          image: ddysher/dns-backend
+          image: gcr.io/google_samples/dns-backend:v1
```
Comment (brendandburns) [inline examples/cluster-dns/dns-backend-rc.yaml]: change this to `gcr.io/google_containers/example-dns-backend:v1` this image now exists.

File: examples/cluster-dns/dns-frontend-pod.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ metadata:
 spec:
   containers:
     - name: dns-frontend
-      image: ddysher/dns-frontend
+      image: gcr.io/google_samples/dns-frontend:v1
```
Comment (brendandburns) [inline examples/cluster-dns/dns-frontend-pod.yaml]: change this to `gcr.io/google_containers/example-dns-frontend:v1` this image now exists.

File: examples/cluster-dns/images/frontend/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM python:2.7
+FROM python:2.7-slim
```
Comment (brendandburns) [inline examples/cluster-dns/images/frontend/Dockerfile]: should we base the backend off of this too?

**Comment by brendandburns** [comment]
Can you add Makefiles for making and pushing the two images?

Thanks!
--brendan

File: examples/cluster-dns/images/frontend/Dockerfile (hunk 3 above)
Comment (sttts) [inline examples/cluster-dns/images/frontend/Dockerfile]: I will check whether it's also happy with the slim version.

**Comment by sttts** [comment]
@brendandburns addressed all your comments.

**Pull request merged**